Ghidra/Features/PDB/src/pdb/cpp/pdb.cpp

```diff
@@ -51,14 +52,16 @@ int init(const char * szFilename, const char * szSignature, const char * szAge)
 				break;
 			default:
 				char msg[256];
-				sprintf(msg, "Unspecified error occurred: 0x%x\n", hr);
+				sprintf_s(msg, 256, "Unspecified error occurred: 0x%lx\n", hr);
```

Comment:
these _s functions are MSVC specific - shouldn't this code be possible to build with other compilers?

Reply:
Hi @paulsapps,

Will respond to both of your comments here (maybe bad form but....).

Re unique_ptr/vector, yes, maybe, but honestly there are **so many** ways this code could be improved and frankly, per @nihilus's comment, the time would be better invested developing a Java-based capability for parsing PDBs based on Microsoft's recently open-sourced spec.  I am among the guilty parties responsible for this code and clearly never envisioned it surviving so long.  Fixing the security holes is obviously the least I can do, but possibly the most I am eager to do.

A Java-based effort would also address the more general platform issue, allowing the ingest of PDB information into Ghidra on non-Windows OSes.  I don't really see much upside to building this code with other compilers.  After all, pdb.exe is merely a wrapper for msdia###.dll, a Microsoft project developed and compiled with MSVC.  Porting the requisite header files containing the msdia structures, error messages, etc. seems an interesting but unproductive exercise.  Of course, willing to hear arguments to the contrary - may be just my laziness talking.  :)

D

Reply:
if its dead code walking then it seems like a fair argument I suppose 🙉